test/vstest.console.UnitTests/InProcessVsTestConsoleWrapperTests.cs

```diff
@@ -114,6 +116,55 @@ public void InProcessWrapperConstructorShouldSetEnvironmentVariablesReceivedAsCo
         _mockEnvironmentVariableHelper.Verify(evh => evh.SetEnvironmentVariable(environmentVariableName, "1"));
     }
 
+    [TestMethod]
+    public void InProcessWrapperConstructorShouldSetEnvironmentVariablesReceivedAsConsoleParametersForProcessHelper()
+    {
+        const string environmentVariableName = "AAAAA";
+
+        var consoleParams = new ConsoleParameters();
+        consoleParams.EnvironmentVariables.Add(environmentVariableName, "1");
+
+        var _ = new InProcessVsTestConsoleWrapper(
+            consoleParams,
+            _mockEnvironmentVariableHelper.Object,
+            _mockRequestSender.Object,
+            _mockTestRequestManager.Object,
+            new Executor(_mockOutput.Object, new Mock<ITestPlatformEventSource>().Object, new ProcessHelper(), new PlatformEnvironment()),
+            new Mock<ITestPlatformEventSource>().Object);
+
+        Assert.IsTrue(ProcessHelper.ExternalEnvironmentVariables?.ContainsKey(environmentVariableName));
+        Assert.IsTrue(ProcessHelper.ExternalEnvironmentVariables?[environmentVariableName] == "1");
```

Comment:
This should make sure there are also the other env variables, because right now there probably is just 1, but there should be all of them, with AAAA added.

Reply:
Do you mean the final set of environment variables that is used to spawn the testhost should contain, among others, `AAAAA` as well ? Because unfortunately we cannot test that, we don't have access to the process start info due to the way `ProcessHelper` is architectured. I'm just making sure the in-process wrapper correctly sets the console params env variables for the `ProcessHelper` to use, so that we don't regress.

Reply:
The test is not testing if inherit env variables is used correctly. If you pass AAAA=1, then in both cases checking AAAA=1  will be true. But you don't know anything about whether AAAA=1 is now the only env variables in place (inherit=false), or if AAAA=1 was added to the existing set of variables (inherit=true).

Reply:
Fixed, thanks !